sqlglot/parser.py

```diff
@@ -2005,6 +2005,20 @@ def _parse_with_property(self) -> t.Optional[exp.Expression] | t.List[exp.Expres
         if self._match(TokenType.SERDE_PROPERTIES, advance=False):
             return self._parse_serde_properties(with_=True)
 
+        if self._match_text_seq("READ", "ONLY") or self._match_text_seq("CHECK", "OPTION"):
+            # Oracle <subquery_restriction_clause>
+            klass = (
+                exp.WithReadOnlyProperty
+                if self._prev.text.upper() == "ONLY"
+                else exp.WithCheckOptionProperty
+            )
+            this = (
+                self._parse_constraint()
+                if self._match(TokenType.CONSTRAINT, advance=False)
+                else None
+            )
```

Comment:
This needs to be tested

Reply:
This actually seems to be in the form of `CONSTRAINT <identifier>`, per the docs:

```
CONSTRAINT constraint

Specify the name of the CHECK OPTION constraint. If you omit this identifier, then Oracle automatically assigns the constraint a name of the form SYS_Cn, where n is an integer that makes the constraint name unique within the database.
```

So I'll also change the parsing here